plugins/providers/virtualbox/action/prepare_clone_snapshot.rb

```diff
@@ -0,0 +1,65 @@
+require "log4r"
+
+require "digest/md5"
+
+module VagrantPlugins
+  module ProviderVirtualBox
+    module Action
+      class PrepareCloneSnapshot
+        def initialize(app, env)
+          @app = app
+          @logger = Log4r::Logger.new("vagrant::action::vm::prepare_clone")
+        end
+
+        def call(env)
+          if !env[:clone_id]
+            @logger.info("no clone master, not preparing clone snapshot")
+            return @app.call(env)
+          end
+
+          # If we're not doing a linked clone, snapshots don't matter
+          if !env[:machine].provider_config.linked_clone
+            return @app.call(env)
+          end
+
+          # We lock so that we don't snapshot in parallel
+          lock_key = Digest::MD5.hexdigest("#{env[:clone_id]}-snapshot")
+          env[:machine].env.lock(lock_key, retry: true) do
+            prepare_snapshot(env)
+          end
+
+          # Continue
+          @app.call(env)
+        end
+
+        protected
+
+        def prepare_snapshot(env)
+          name = env[:machine].provider_config.linked_clone_snapshot
+          name_set = !!name
+          name = "base" if !name
```

Comment:
@mitchellh @sethvargo 
`"base"` - is it some convention? Will Otto refer to this snapshot by this explicit name?

I'm asking because in "parallels" provider we have a default snapshot name `"vagrant_linked_clone"`. So I'd like to understand - should I change it to "base" or not. 
Thanks.